django/utils/log.py

```diff
@@ -189,3 +191,41 @@ def format(self, record):
 
         record.msg = msg
         return super(ServerFormatter, self).format(record)
+
+
+def log_response(message, *args, **kwargs):
+    """Log errors based on response status.
+
+    Log 5xx responses in error level and 4xx in warning level (unless level
+    is explicitly given as a keyword argument). The Response status_code and
+    the entire request are passed to logger extra parameter. This function
+    also avoid duplicated logs.
+
+    """
+
+    response = kwargs.get('response')
+    request = kwargs.get('request', None)
+    logger = kwargs.get('logger', request_logger)
+    level = kwargs.get('level', None)
+    exc_info = kwargs.get('exc_info', None)
+
+    # Check if the response has already been logged
```

Comment:
I think it may be worth expanding this comment slightly to explain _why_ `log_response` may be called multiple times on the same response (that is, that we want to log with `exc_info` when we can, and we need to do that right where we're converting an exception to a response), but we also want a catch-all later on to make sure that all 4xx and 5xx responses are logged (even if they did not originate from a caught exception).